sfs/time/drivingfunction.py

```diff
@@ -1,64 +1,197 @@
+    xref = util.asarray_1d(xref)
+    g0 = np.sqrt(2 * np.pi * np.linalg.norm(xref - x0, axis=1))
     ds = x0 - xs
-    r = np.linalg.norm(x0-xs, axis=1)
-    delay = 1/c * r
-    weight = g0/(2*np.pi) * inner1d(ds, nx0) / r**(3/2)
-    sig = np.ones(len(x0))  # dirac
-    line = delayline(sig, delay*fs, weight)
-    return delay, weight, line
+    r = np.linalg.norm(x0 - xs, axis=1)
+    delays = r/c
+    weights = g0/(2 * np.pi) * inner1d(ds, n0) / r**(3/2)
+    return delays, weights
+
+
+def driving_signals(delays, weights, signal, fs=None):
+    """Get driving signals per secondary source.
+
+    Returned signal is the delayed and weighted input signal per channel.
+
+    Parameters
+    ----------
+    delays : (C,) array_like
+        Delay in seconds for each channel, negative values allowed.
+    weights : (C,) array_like
+        Amplitude weighting factor for each channel.
+    signal : (N,) array_like
+        Excitation signal (mono) which gets weighted and delayed.
+    fs: int, optional
+        Sampling frequency in hertz.
+
+    Returns
+    -------
+    (N, C) numpy.ndarray
+        N is signal samples(t) per channel C, delayed and weighted.
+    t_offset : float
+        Simulation point in time offset (seconds).
+
+    """
+    delays = util.asarray_1d(delays)
+    weights = util.asarray_1d(weights)
+    d, t_offset = apply_delays(signal, delays, fs)
+    return d * weights, t_offset
 
 
-def delayline(sig, delay, weight):
-    """Delayline with weights, no fractional delay yet.
+def apply_delays(signal, delays, fs=None):
+    """Apply Delays for every channel.
 
-    Delay in samples.
-    sig (m x n) where columns n are channels and rows m the corresponding
-    signals.
-    Weight is array with weighting for each channel.
+    Delay in seconds, no fractional delay. An absolute time offset is applied.
 
-    """
-    delay = np.rint(delay).astype(int)  # no fractional delay, samples
-    extention = np.max(delay)  # avoid cutting signal
-    channels = len(weight)  # extract no. of channels
-    sig_length = sig.ndim
-    out = np.zeros([sig_length+extention, channels])  # create shape
+    Parameters
+    ----------
+    signal : (N,) array_like
+        Excitation signal (mono) which gets weighted and delayed.
+    delay : (C,) array_like, in seconds
+        delay in seconds for each channel, negative values allowed.
+    fs: int, optional
+        Sampling frequency in hertz.
 
-    for channel in range(channels):
-        cdelay = delay[channel]  # get channel delay
-        out[cdelay:cdelay+sig_length, channel] = sig[channel] * weight[channel]
+    Returns
+    -------
+    out : (N, C) numpy.ndarray
+        N is signal samples(t) per channel C.
+    t_offset : float
+        Simulation point in time offset (seconds).
 
-    return out
+    """
+    if fs is None:
+        fs = defs.fs
+    signal = util.asarray_1d(signal)
+    delays = util.asarray_1d(delays)
+
+    # convert to samples
```

Comment:
Since the word `samples` is very frequently used in the following few lines, is this comment really necessary?